tests/unit/lcos/channel.cpp

```diff
@@ -0,0 +1,79 @@
+//  Copyright (c) 2015 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_init.hpp>
+#include <hpx/include/lcos.hpp>
+#include <hpx/include/local_lcos.hpp>
+#include <hpx/util/lightweight_test.hpp>
+
+#define NUM_VALUES 100
+
+void send_values(hpx::lcos::local::channel<int> &c)
+{
+    for (int i = 0; i != NUM_VALUES; ++i)
+    {
+        c.set_value(i);
+    }
+}
+
+void receive_values(hpx::lcos::local::channel<int> &c)
+{
+    for (int i = 0; i != NUM_VALUES; ++i)
+    {
+        hpx::future<int> f = c.get_future();
+        HPX_TEST_EQ(i, f.get());
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+int hpx_main()
+{
+    {
+        hpx::lcos::local::channel<int> c;
+
+        hpx::future<void> g = hpx::async(&send_values, boost::ref(c));
+        receive_values(c);
+
+        g.get();
+    }
+
+    {
+        hpx::lcos::local::channel<int> c;
+
+        hpx::future<void> g = hpx::async(&receive_values, boost::ref(c));
+        send_values(c);
+
+        g.get();
+    }
+
+    // two calls to get_future() without waiting for the first one to become
+    // ready should throw
```

Comment:
I'm not sure if that is something that's practical. Consider something like that:
```
while(...)
{
    c.get_future().then(...);
}
```

With that restriction, I think code like that is not safe.